### PR TITLE
expand ci pipeline

### PR DESCRIPTION
**Pull request by seanstory**
This version of the CI pipeline actually attaches artifacts and parses the test output. It _does_ required the [Pipeline Maven Integration](https://plugins.jenkins.io/pipeline-maven/#installation) plugin to be installed.

### CODE REVIEW

**Comment by tarekziade** [comment]
If you want to keep the packed version, you can use also the `script` type like this:

```
@Library(['estc', 'entsearch']) _

eshPipeline(
    timeout: 45,
    project_name: 'Thumbnails4j',
    repository: 'thumbnails4j',
    stage_name: 'Thumbnails4j Unit Tests',
    stages: [
      [
          name: 'Maven Build',
          type: 'script',
          label: 'Maven Build',
          script: {
              withMaven {
                  sh './mvnw clean verify'
              }
           }.
          match_on_all_branches: true,
      ]
    ],
    slack_channel: 'workplace-search-connectors'
)
```

`script` is a callable where you can write your own sequence, unlike `sh`

The benefit of this approach is that you keep extra features we have in our lib like slack integration.

File: .ci/pipelines/thumbnails4j.groovy
Comment (tarekziade) [inline .ci/pipelines/thumbnails4j.groovy:27]: If you use the plain pipeline sequence, you might need to provide the `options` and `parameters` parts (see ent-search packaging's pipeline as an example)

**Review by tarekziade — DISMISSED** [review]
LGTM -- pending a test with the plugin

**Comment by seanstory** [comment]
@tarekziade any idea why the `Thumbnails4j` "check" on this PR is showing as green, even though it has consistently failed on CI? https://swiftype-ci.elastic.co/job/thumbnails4j/job/thumbnails4j/job/seanstory%252Fexpand-ci-pipeline/

**Comment by tarekziade** [comment]
@seanstory that would be a bug in my lib. for some reason, the error is not bubbling up. Will investigate, thanks

**Comment by tarekziade** [comment]
"Yay" it breaks properly now :)

@seanstory Once you've reviewed my lib patch I can release it to stable and this branch can get back to it by removing my change

**Comment by seanstory** [comment]
@elastic/workplace-search-connectors can I get a review/approval on this?

**Review by RossAllenBell — APPROVED** [review]

**Pull request merged**
